website/components/featured-slider/index.jsx

```diff
@@ -0,0 +1,221 @@
+import React, { Component } from 'react'
+import StatusBar from './StatusBar'
+import marked from 'marked'
+import Button from '@hashicorp/react-button'
+import Image from '@hashicorp/react-image'
+
+class FeaturedSlider extends Component {
```

Comment:
This looks pretty similar to our [`case-study-slider`](https://github.com/hashicorp/web-components/tree/master/packages/case-study-slider) component - can we use it here instead of duplicating? (I haven't diffed the code, and you probably have a great reason for doing this, but wanted to just confirm)

If it's necessary, might be worth dropping a comment here with a link back to the component source.

Reply:
Yeah, this is a good callout here and am glad you brought it up.  So there was actually quite a bit of deep thought that went into this one here, so I'll try to walk you through it:

## The Problem

So in short, the prop structure of the `case-study-slider` is strongly coupled to an assumption of **what** the content is, and made several strong contextual assumptions about the behavior.  These assumptions broke down in the context of how this is used on the Nomad site here.  There were a bunch of small behaviors / assumptions that broke down that I need to fix, but I'll give one concrete example here that will make it easier to grok:

Previously there was a [caseStudyResource.slug](https://github.com/hashicorp/web-components/blob/master/packages/case-study-slider/index.js#L180) prop that alone powered the link of the button: to an inbound relative link (`/resources/{case-study-slug}`). 
For obvious reasons those relative links wouldn't work because we are on a new site, but I couldn't even just replace it to be a direct link to HashiCorp.com because we aren't always linking to `hashicorp.com/resources/{case-study-slug}` (a concrete example being we link to https://www.hashicorp.com/case-studies/roblox/)

## The Solution

Rather than breaking the interface of the `case-study-slider` (or introducing a questionable bandage which would complicate that model), I reworked the `case-studies-slider` to make the required adjustments, but also spent time decoupling it from the notion of case studies and I renamed it to `featured-slider` to represent that it's no longer coupled to case studies, and it's just a visual pattern.

I implemented this directly into a NomadProject.io component for purely expediency purposes.  My general recommendation here is that we deprecate the `case-studies-slider`, and elevate the `featured-slider` to a web component, as the view-model allows us to use this element in different contexts.

I actually made a [props.js](https://github.com/hashicorp/nomad/blob/3c306ef2d3f4c2a8068637651b50aedac90595b8/website/components/featured-slider/index.props.js
) file for this component because I had this in mind as I was building it.

## Why there's no comment

I actually thought about this pretty hard of if I should have a comment or not, and leaned towards no.  Totally can be swayed the other way here, but here's my reasoning:

If you look at the code by itself, there's no need to provide context of why we're not using a private component in a public repo; and it would only confuse any OSS users scanning the codebase. This as organizational debt, and not actual code debt. It needs to be documented, but not in the codebase.

I already have this slated to be a discussion in our web-components meeting & I will point directly to this comment here (to thank you for giving me the opportunity here to explain this!)

Reply:
This is great. Thank you for walking me through the thinking here. 💯 agree.